src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/GlobalSearchService.kt

```diff
@@ -61,6 +66,103 @@ class GlobalSearchService(
     }
   }
 
+  data class NomisContext(var nomisFinished: Boolean, var num: String?)
+
+  fun compareIndex(): Pair<List<String>, List<String>> {
+    val onlyInIndex = mutableListOf<String>()
+    val onlyInNomis = mutableListOf<String>()
+
+    val allNomis =
+      prisonerIndexService.getAllNomisOffenders(0, Int.MAX_VALUE).offenderIds!!.sortedBy { it.offenderNumber }
+    val iter = allNomis.iterator()
+
+    val nomisContext =
+      NomisContext(
+        nomisFinished = !iter.hasNext(),
+        num = if (iter.hasNext()) iter.next().offenderNumber else null
+      )
+
+    val scroll = Scroll(TimeValue.timeValueMinutes(1L))
+    var searchResponse = setupIndexSearch(scroll)
+
+    var scrollId = searchResponse.scrollId
+    var searchHits = searchResponse.hits.hits
+
+    while (ArrayUtils.isNotEmpty(searchHits)) {
+      val allIndex = searchHits.map { it.sourceAsMap["prisonerNumber"] as String }
+
+      processIndexScroll(allIndex, nomisContext, onlyInIndex, onlyInNomis, iter)
+
+      val scrollRequest = SearchScrollRequest(scrollId)
+      scrollRequest.scroll(scroll)
+      searchResponse = searchClient.scroll(scrollRequest)
+
+      scrollId = searchResponse.scrollId
+      searchHits = searchResponse.hits.hits
+    }
+    processIndexEnd(nomisContext, onlyInNomis, iter)
+
+    return Pair(onlyInIndex, onlyInNomis)
+  }
+
+  private fun setupIndexSearch(scroll: Scroll): SearchResponse {
+
+    val searchSourceBuilder = SearchSourceBuilder().apply {
+      docValueField("prisonerNumber")
+      sort("prisonerNumber")
+      size(10000)
+    }
+    val searchRequest = SearchRequest(arrayOf(getIndex()), searchSourceBuilder)
+    searchRequest.scroll(scroll)
+    return searchClient.search(searchRequest)
+  }
+
+  private fun processIndexScroll(
+    allIndex: List<String>,
+    nomisContext: NomisContext,
+    onlyInIndex: MutableList<String>,
+    onlyInNomis: MutableList<String>,
+    iter: Iterator<OffenderId>
```

Comment:
This all seems a bit complicated, manually controlling iterators. Can't we just use [minus](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/minus.html)?